sql/sql_show.cc

```diff
@@ -1267,8 +1269,27 @@ mysqld_show_create(THD *thd, TABLE_LIST *table_list)
   */
   MDL_savepoint mdl_savepoint= thd->mdl_context.mdl_savepoint();
 
+  const char *orig_name= NULL;
+  TABLE_LIST tl;
+  bool versioned_query=
+      table_list->vers_conditions.type != FOR_SYSTEM_TIME_UNSPECIFIED;
+  String archive_name;
+  if (versioned_query)
+  {
+    DBUG_ASSERT(table_list->vers_conditions.type == FOR_SYSTEM_TIME_AS_OF);
+    VTMD_table vtmd(*table_list);
+    if (vtmd.find_archive_name(thd, archive_name))
+      goto exit;
+
+    tl.init_one_table(table_list->db, table_list->db_length, archive_name.ptr(),
+                      archive_name.length(), archive_name.ptr(), TL_READ);
+
+    orig_name= table_list->table_name;
+    table_list= &tl;
+  }
 
-  if (mysqld_show_create_get_fields(thd, table_list, &field_list, &buffer))
+  if (mysqld_show_create_get_fields(thd, table_list, &field_list, &buffer,
+                                    orig_name))
```

Comment:
Why do you changed interface of `mysqld_show_create_get_fields()`? Was it required? If you can modify existing parameters, don't touch interface. Look at code from `show_create_table()` I put in #125 to learn how to set parameters properly.